addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -136,6 +136,12 @@ func (wh *ClusterBootstrap) Default(ctx context.Context, obj runtime.Object) err
 	// Get the helper ready for the defaulting logic
 	helper := clusterbootstrapclone.Helper{Logger: clusterbootstraplog}
 
+	_, usingCustomCNIProvider := clusterBootstrap.Annotations[constants.UseCustomCNIProvider]
+
+	if usingCustomCNIProvider && clusterBootstrap.Spec.CNI != nil {
```

Comment:
Added to both default and validator because if the user submits a clusterbootstrap with the AddCBMissingFieldsAnnotationKey and UnmanagedCNI annotations and the spec.cni != nil, it is difficult to interpret since UnmangeCNI means spec.cni = nil, but the user did provide one.
We could make the decision for user and simply set spec.cni = nil whenever we see unmanagecni regardless of the user provided value for spec.cni

Reply:
we will move ahead with setting spec.nil == nil if clusterbootstrap is annotated. (ignoring user provided spec.cni because this is is a default webhook, so we will take the annotation presence as a request of user to fill in spec.cni for them).

Reply:
done